core/src/main/java/cucumber/runtime/ThreadLocalRunnerSupplier.java

```diff
@@ -38,7 +42,7 @@ public Runner get() {
     }
 
     private Runner createRunner() {
-        return new Runner(glueSupplier.get(), eventBus, backendSupplier.get(), runtimeOptions);
+        return new Runner(glueSupplier.get(), new TestCaseEventBus(eventBus), backendSupplier.get(), runtimeOptions);
```

Comment:
This solves the JUnit problem.

Reply:
But for some tests, cannot remember which, I needed to specifically pass in the test `EventBus` implementation and it had registrations for `TestStepStarted` that needed to be sent, not queued, but your calls to `parent.send` or `super.send` may also resolved.  Can’t quite tell as am reviewing on web page rather than in IDE

Reply:
You are entirely right. The failing tests were the formatter tests, specifically the ones with timestamps.

I initially resolved that by not using the runtime builder and passing the event bus directly into the runner. I figured this was okay because we're testing the formatter, not the whole integration. The runtime was used to generate a series of events, nothing else. This means the bus doesn't need to be wrapped. There is no parallel execution.

Turns out this was rather useful so I changed the runtime builder to run on the main thread directly when threads were not needed.